src/gsi/CMakeLists.txt

```diff
@@ -155,7 +155,11 @@ target_link_libraries(gsi_fortran_obj PUBLIC nemsio::nemsio)
 target_link_libraries(gsi_fortran_obj PUBLIC ncio::ncio)
 target_link_libraries(gsi_fortran_obj PUBLIC w3emc::w3emc_d)
 target_link_libraries(gsi_fortran_obj PUBLIC sp::sp_d)
-target_link_libraries(gsi_fortran_obj PUBLIC bufr::bufr_d)
+if(DEFINED ENV{USE_BUFR4})
```

Comment:
Looks OK to me.  Cross checking with @DavidHuber-NOAA .  PR #791 upgrades to bufr/12.1.0.  Not sure how the bufr logic added here might impact Dave's PR.